[
  {
    "project": "encode/httpx",
    "commit": "8fd5b71016c10dd266a058966f3ff3843e3ce9c6",
    "filename": "httpx/_client.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/encode-httpx/httpx/_client.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "httpx/_client.py:1905:8 Incompatible variable type [9]: exc_type is declared to have type `typing.Type[BaseException]` but is used as type `None`.",
    "message": " exc_type is declared to have type `typing.Type[BaseException]` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 1905,
    "warning_line": "        exc_type: typing.Type[BaseException] = None,"
  },
  {
    "project": "encode/httpx",
    "commit": "8fd5b71016c10dd266a058966f3ff3843e3ce9c6",
    "filename": "httpx/_client.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/encode-httpx/httpx/_client.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "httpx/_client.py:1906:8 Incompatible variable type [9]: exc_value is declared to have type `BaseException` but is used as type `None`.",
    "message": " exc_value is declared to have type `BaseException` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 1906,
    "warning_line": "        exc_value: BaseException = None,"
  },
  {
    "project": "encode/httpx",
    "commit": "8fd5b71016c10dd266a058966f3ff3843e3ce9c6",
    "filename": "httpx/_client.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/encode-httpx/httpx/_client.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "httpx/_client.py:1907:8 Incompatible variable type [9]: traceback is declared to have type `TracebackType` but is used as type `None`.",
    "message": " traceback is declared to have type `TracebackType` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 1907,
    "warning_line": "        traceback: TracebackType = None,"
  },
  {
    "project": "encode/httpx",
    "commit": "8fd5b71016c10dd266a058966f3ff3843e3ce9c6",
    "filename": "httpx/_client.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/encode-httpx/httpx/_client.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "httpx/_client.py:1927:8 Incompatible variable type [9]: exc_type is declared to have type `typing.Type[BaseException]` but is used as type `None`.",
    "message": " exc_type is declared to have type `typing.Type[BaseException]` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 1927,
    "warning_line": "        exc_type: typing.Type[BaseException] = None,"
  },
  {
    "project": "encode/httpx",
    "commit": "8fd5b71016c10dd266a058966f3ff3843e3ce9c6",
    "filename": "httpx/_client.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/encode-httpx/httpx/_client.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "httpx/_client.py:1928:8 Incompatible variable type [9]: exc_value is declared to have type `BaseException` but is used as type `None`.",
    "message": " exc_value is declared to have type `BaseException` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 1928,
    "warning_line": "        exc_value: BaseException = None,"
  },
  {
    "project": "encode/httpx",
    "commit": "8fd5b71016c10dd266a058966f3ff3843e3ce9c6",
    "filename": "httpx/_client.py",
    "filename_after_commit": "/home/chowyi/TypeAnnotation_Study/GitHub/encode-httpx/httpx/_client.py",
    "file_hunks_size": 6,
    "min_patch_found": false,
    "single_hunk": false,
    "fit_TFix": false,
    "delete_only_patch": false,
    "has_suppression_all_hunks": false,
    "full_warning_msg": "httpx/_client.py:1929:8 Incompatible variable type [9]: traceback is declared to have type `TracebackType` but is used as type `None`.",
    "message": " traceback is declared to have type `TracebackType` but is used as type `None`.",
    "rule_id": "Incompatible variable type [9]",
    "warning_line_no": 1929,
    "warning_line": "        traceback: TracebackType = None,"
  }
]